SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslationManager.cpp

```diff
@@ -130,9 +175,74 @@ void SpatialVirtualWorkerTranslationManager::SendVirtualWorkerMappingUpdate() co
 	Connection->SendComponentUpdate(SpatialConstants::INITIAL_VIRTUAL_WORKER_TRANSLATOR_ENTITY_ID, &Update);
 }
 
+void SpatialVirtualWorkerTranslationManager::SpawnPartitionEntity(Worker_EntityId PartitionEntityId, VirtualWorkerId VirtualWorkerId)
+{
+	TArray<FWorkerComponentData> Components = SpatialGDK::EntityFactory::CreatePartitionEntityComponents(
+		PartitionEntityId, Translator->NetDriver->InterestFactory.Get(), Translator->LoadBalanceStrategy.Get(), VirtualWorkerId);
+
+	const Worker_RequestId RequestId = Connection->SendCreateEntityRequest(MoveTemp(Components), &PartitionEntityId);
+
+	CreateEntityDelegate OnCreateWorkerEntityResponse;
+	OnCreateWorkerEntityResponse.BindLambda([this, VirtualWorkerId](const Worker_CreateEntityResponseOp& Op) {
+		if (Op.status_code == WORKER_STATUS_CODE_SUCCESS)
+		{
+			UE_LOG(LogSpatialVirtualWorkerTranslationManager, Log,
+				   TEXT("Successfully created partition entity. "
+						"Entity: %lld. Virtual Worker: %d"),
+				   Op.entity_id, VirtualWorkerId);
+			OnPartitionEntityCreation(Op.entity_id, VirtualWorkerId);
+			return;
+		}
+
+		if (Op.status_code != WORKER_STATUS_CODE_TIMEOUT)
+		{
+			UE_LOG(LogSpatialVirtualWorkerTranslationManager, Error,
+				   TEXT("Partition entity creation failed: \"%s\". "
+						"Entity: %lld. Virtual Worker: %d"),
+				   UTF8_TO_TCHAR(Op.message), Op.entity_id, VirtualWorkerId);
+			return;
+		}
+
+		UE_LOG(LogSpatialVirtualWorkerTranslationManager, Warning,
+			   TEXT("Retrying timed out partition entity creation. "
+					"Entity: %lld. Virtual Worker: %d"),
+			   Op.entity_id, VirtualWorkerId);
+
+		SpawnPartitionEntity(Op.entity_id, VirtualWorkerId);
+	});
+
+	Receiver->AddCreateEntityDelegate(RequestId, MoveTemp(OnCreateWorkerEntityResponse));
+}
+
+void SpatialVirtualWorkerTranslationManager::OnPartitionEntityCreation(Worker_EntityId PartitionEntityId, VirtualWorkerId VirtualWorker)
+{
+	Partitions.Emplace(PartitionInfo{ PartitionEntityId, VirtualWorker, SpatialConstants::INVALID_ENTITY_ID });
+
+	UE_LOG(LogSpatialVirtualWorkerTranslationManager, Log,
+		   TEXT("Adding translation manager mapping. Virtual worker %d -> Parition entity %lld"), VirtualWorker, PartitionEntityId);
+
+	if (Partitions.Num() == VirtualWorkersToAssign.Num())
+	{
+		UE_LOG(LogSpatialVirtualWorkerTranslationManager, Log,
+			   TEXT("Found all %d required partitions, querying for server worker entities"), VirtualWorkersToAssign.Num());
+		QueryForServerWorkerEntities();
+	}
+	else
+	{
+		UE_LOG(LogSpatialVirtualWorkerTranslationManager, Log,
+			   TEXT("Didn't find all %d required partitions, only found %d, currently have:"), VirtualWorkersToAssign.Num(),
+			   Partitions.Num());
+		for (const PartitionInfo& Partition : Partitions)
+		{
+			UE_LOG(LogSpatialVirtualWorkerTranslationManager, Log, TEXT(" - virtual worker %d -> partition entity %lld"),
+				   Partition.VirtualWorker, Partition.PartitionEntityId);
+		}
```

Comment:
Do we expect to print this for every partition until we create all of the corresponding entities? Seems a bit spammy.

Reply:
Switch to verbose?

Reply:
Done